src/main/scala/akka/persistence/jdbc/query/JournalSequenceActor.scala

```diff
@@ -136,18 +136,18 @@ class JournalSequenceActor(readJournalDao: ReadJournalDao, config: JournalSequen
               currentElement
             } else currentMax
 
+          // we accumulate in newMissing the gaps we detect on each iteration
+          val newMissing =
+            if (previousElement + 1 == currentElement || newMax == currentElement) missing
+            else missing.addRange(previousElement + 1, currentElement)
```

Comment:
Based on the previous PR comment I managed to simplify this piece of code. I did have to add the `|| newMax == currentElement` condition though (otherwise some elements which were part of givenUp were included in the missing elements collection).